tests/framework/extensions/rbac/verify.go

```diff
@@ -0,0 +1,253 @@
+package rbac
+
+import (
+	"fmt"
+	"net/url"
+	"regexp"
+	"strings"
+	"testing"
+
+	apiV1 "github.com/rancher/rancher/pkg/apis/provisioning.cattle.io/v1"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	management "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+	v1 "github.com/rancher/rancher/tests/framework/clients/rancher/v1"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters"
+	"github.com/rancher/rancher/tests/framework/extensions/namespaces"
+	"github.com/rancher/rancher/tests/framework/extensions/projects"
+	"github.com/rancher/rancher/tests/framework/extensions/users"
+	namegen "github.com/rancher/rancher/tests/framework/pkg/namegenerator"
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	coreV1 "k8s.io/api/core/v1"
+)
+
+const (
+	roleOwner                     = "cluster-owner"
+	roleMember                    = "cluster-member"
+	roleProjectOwner              = "project-owner"
+	roleProjectMember             = "project-member"
+	roleCustomManageProjectMember = "projectroletemplatebindings-manage"
+	roleCustomCreateNS            = "create-ns"
+	roleProjectReadOnly           = "read-only"
+	restrictedAdmin               = "restricted-admin"
+	standardUser                  = "user"
+	activeStatus                  = "active"
+	forbiddenError                = "403 Forbidden"
+)
+
+// VerifyGlobalRoleBindings validates that a global role bindings is created for a user when the user is created
+func VerifyGlobalRoleBindings(t *testing.T, user *management.User, adminClient *rancher.Client) {
+	query := url.Values{"filter": {"userName=" + user.ID}}
+	grbs, err := adminClient.Steve.SteveType("management.cattle.io.globalrolebinding").List(query)
+	require.NoError(t, err)
+	assert.Equal(t, 1, len(grbs.Data))
+}
+
+// VerifyListCluster validates a user with the required global permissions are able to/not able to list the clusters in rancher server
+func VerifyListCluster(t *testing.T, client, standardClient *rancher.Client, clusterID, role string) {
+	clusterList, err := standardClient.Steve.SteveType(clusters.ProvisioningSteveResourceType).ListAll(nil)
+	require.NoError(t, err)
+	clusterStatus := &apiV1.ClusterStatus{}
+	err = v1.ConvertToK8sType(clusterList.Data[0].Status, clusterStatus)
```

Comment:
What would be the first item of the cluster list, local or the first downstream? Would this indexing affect the test behavior if we have multiple clusters environment?

Reply:
I still don't understand, what if there is a lot of downstream clusters. This seems like you're only assuming that there is one downstream cluster. Am I wrong Anu?

Reply:
We create a standard user and add that user in the downstream cluster thats configured in rancher server.  Even if there are a lot of downstream clusters, as long as the user is not added in any of them [which we are not doing in our tests], they won't be able to list the downstream clusters as the user is a standard user and we are getting the list of clusters as that standard user. let me know if that makes sense.

Reply:
are you saying that clusterList.Data[0] is always local cluster, then?

Reply:
It's always the downstream cluster the standard user has access to.